File: web/di_core/domain/Model/User/UserFullDetailInfo.ts
import { UserInfo } from '@core/domain/Model/User/UserInfo';
import { UserProfile } from '@core/domain/Model/User/UserProfile';

export class UserFullDetailInfo {
  user: UserInfo;
  profile?: UserProfile;

  constructor(user: UserInfo, profile?: UserProfile) {
    this.user = user;
    this.profile = profile;
  }

  static fromObject(object: any): UserFullDetailInfo {
    return new UserFullDetailInfo(UserInfo.fromObject(object.user), UserProfile.fromObject(object.profile ?? {}));
  }
}

File: server/src/main/scala/co/datainsider/caas/user_profile/domain/profile/UserFullDetailInfo.scala
package co.datainsider.caas.user_profile.domain.profile

import co.datainsider.caas.user_caas.domain.UserGroup.UserGroup
import co.datainsider.caas.user_caas.domain.UserGroupRef
import co.datainsider.caas.user_profile.domain.user.{UserInfo, UserProfile}
import com.fasterxml.jackson.module.scala.JsonScalaEnumeration

/**
  * @author anhlt
  */
case class UserFullDetailInfo(
    user: UserInfo,
    profile: Option[UserProfile],
    @JsonScalaEnumeration(classOf[UserGroupRef])
    userGroup: Option[UserGroup] = None
)
